src/model/document.js

```diff
@@ -278,6 +322,46 @@ export default class Document {
 		return null;
 	}
 
+	/**
+	 * Used to register a post-fixer callback. Post-fixers mechanism guarantees that the features that listen to
+	 * {@link module:engine/model/model~Model#event:_change model's change event} will operate on a correct model state.
+	 *
+	 * Execution of a feature may lead to an incorrect document tree state. The callbacks are used to fix document tree after
+	 * it has changed. Post-fixers are fired just after all changes from the outermost change block were applied but
+	 * before {@link module:engine/model/document~Document#event:change} is fired. If a post-fixer callback made a change,
+	 * it should return `true`. When this happens, all post-fixers are fired again to check if something else should
+	 * not be fixed in the new document tree state.
+	 *
+	 * As a parameter, a post-fixer callback receives {@link module:engine/model/writer~Writer} instance connected with the executed
+	 * changes block. Thanks to that, all changes done by the callback will be added to the same {@link module:engine/model/batch~Batch}
+	 * (and undo step) as the original changes. This makes post-fixer changes transparent for the user.
+	 *
+	 * An example of a post-fixer is a callback that checks if all the data was removed from the editor. If so, the
+	 * callback should add an empty paragraph, so that the editor is never empty:
+	 *
+	 *		document.registerPostFixer( writer => {
+	 *			const changes = document.differ.getChanges();
+	 *
+	 *			// Check if the changes lead to an empty root in an editor.
+	 *			let applied = false;
+	 *
+	 *			for ( const entry of changes ) {
+	 *				if ( entry.type == 'remove' && entry.position.root.isEmpty ) {
+	 *					writer.insertElement( 'paragraph', ModelPosition.createAt( entry.position.root, 0 ) );
```

Comment:
You can writer `writer.insertElement( 'paragraph', entry.position.root, 0 );`.

Also since, multi root editor is not very common, you can use `document.getRoot()`

Reply:
I'll leave `entry.position.root` as it is more correct.